src/core/qgsrendercontext.cpp

```diff
@@ -359,3 +384,24 @@ double QgsRenderContext::convertFromMapUnits( double sizeInMapUnits, QgsUnitType
   }
   return 0.0;
 }
+
+double QgsRenderContext::convertMetersToMapUnits( double meters ) const
+{
+  switch ( mDistanceArea.sourceCrs().mapUnits() )
+  {
+    case QgsUnitTypes::DistanceMeters:
+      return meters;
+      break;
+    case QgsUnitTypes::DistanceDegrees:
+    {
+      QgsPoint pointCenter = mExtent.center();
+      pointCenter = mCoordTransform.transform( pointCenter );
```

Comment:
I don't think this is required - the extent will always be in the right CRS here